scala/providers.bzl

```diff
@@ -24,3 +24,33 @@ def create_scala_provider(ijar, class_jar, compile_jars,
       transitive_runtime_jars = transitive_runtime_jars,
       transitive_exports = []  #needed by intellij plugin
   )
+
+ScalacProvider = provider(
+    doc = "ScalaProvider",
+    fields = [
+        "scalac", "scalalib", "scalareflect", "scalacompiler",
```

Comment:
I think going this way vs toolchains i.e. #530 is something we need to think about.

@katre what do you think? This probably lets us sidestep the current toolchain issues blocking us. It also allows us to potentially have more than one build in a single repo (build for both scala 2.11 and 2.12), although I don't see how.

Side note: instead of enumerating the extra jars, can we have something like default_compile_classpath, default_runtime_classpath or something? That way we can control that a bit easier in the different cases (really only macro_library should get reflect by default, and only repl needs scalacompiler in the runtime jars.

Reply:
Re multiple versions in a single repo:
You'd be able to add multiple scala versions (`ScalacProvider`s) in [scala_repositories](https://github.com/bazelbuild/rules_scala/pull/544/files#diff-9957b181001dad28e385437a86e99126R260) with custom labels, e.g. `scala_2_12_6`. Then you could override the default `ScalacProvider` passed to an invocation of `scala_library` etc.

However, this would require some rules (e.g. scala_test) that have dependencies that need to be either 2.11/2.12 to be changed to macros (so they can figure out which versions to use). I haven't come up with a better solution to that problem yet at least

Reply:
@ittaiz re cross building: exactly, I have a version where the main rules are working like that (scala_library, ...) allowing you to build different graphs for different versions.

Decided to create a pr for this simpler version first as getting all rules to work with multiple versions might require some bigger changes